csrc/codegen.cpp

```diff
@@ -786,17 +786,19 @@ class CudaKernelGenerator : private kir::ConstIrVisitor {
     indent() << "}\n";
     auto op_type = rop->getRNGOpType();
     indent() << gen(rop->output(0)) << " = " << op_type;
-    if (needFloatSuffix(op_type) && rop->dtype() == DataType::Float) {
-      code_ << "f";
+    if (needFloatSuffix(op_type)) {
+      if (rop->dtype() == DataType::Float) {
+        code_ << "f";
+      } else if (rop->dtype() == DataType::BFloat16) {
+        code_ << "_bfloat";
+      } else if (rop->dtype() == DataType::Half) {
+        code_ << "_half";
+      }
```

Comment:
This change means we no longer use `gen_uniform_range` i.e. double precision sampling for half precision types, which caused a mismatch with `RNGOp::getPhiloxMultiple()`. I believe this fixes the repeating values problem, so I added a test to try and catch that if it re-occurs.